soh/soh/Enhancements/randomizer/location_access/overworld/kakariko.cpp

```diff
@@ -66,7 +66,7 @@ void RegionTable_Init_Kakariko() {
         Entrance(RR_KAK_WINDMILL,             []{return logic->CanOpenOverworldDoor(RG_WINDMILL_KEY);}),
         Entrance(RR_KAK_BAZAAR,               []{return logic->IsAdult && logic->AtDay && logic->CanOpenOverworldDoor(RG_KAK_BAZAAR_KEY);}),
         Entrance(RR_KAK_SHOOTING_GALLERY,     []{return logic->IsAdult && logic->AtDay && logic->CanOpenOverworldDoor(RG_KAK_SHOOTING_GALLERY_KEY);}),
-        Entrance(RR_KAK_WELL,                 []{return logic->IsAdult || logic->DrainWell || logic->CanUse(RG_IRON_BOOTS);}),
+        Entrance(RR_KAK_WELL,                 []{return logic->IsAdult || logic->DrainWell || logic->CanUse(RG_IRON_BOOTS) || ctx->GetTrickOption(RT_BOTTOM_OF_THE_WELL_NAVI_DIVE) && logic->HasItem(RG_BRONZE_SCALE) && logic->CanJumpslash() && logic->IsChild;}),
```

Comment:
```suggestion
        Entrance(RR_KAK_WELL,                 []{return logic->IsAdult || logic->DrainWell || logic->CanUse(RG_IRON_BOOTS) || (ctx->GetTrickOption(RT_BOTTOM_OF_THE_WELL_NAVI_DIVE) && logic->IsChild && logic->HasItem(RG_BRONZE_SCALE) && logic->CanJumpslash());}),```
Using brackets to isolate && checks from || checks avoids any confusion or bugs from how the code processes them. Also, while a small thing, it's preferred to move the faster checks like IsChild to the start as a minor optimisation.

Reply:
It's good I changed the two tricks

Reply:
you didn't seem to add the brackets

Reply:
refix